### DISCUSSION TITLE
hbridge ic current rating: TB6612FNG

### DISCUSSION THREAD

**Discussion by Cydget** [adamb314/ServoProject#26] [General]
The TB6612FNG is only rated for 1.2A continuous or 3.2A peak current. This shouldn't be an issue for small servos, but for larger ones/poorly manufactured ones may exceed this rating if stalled. Just something to keep in mind if anyone wanted to use this with larger servos, or if you had a new pcb rev in the works.

**Comment by adamb314** [comment]
You are correct that each channel of the `TB6612FGN` chip is rated for 1.2A. But `TB6612FNG` it is a dual channel chip, and I have connected both channels in parallel, so the `combined` PCB  can handle 2.4A continuous and 6.4A peak current. However, I am open to suggestions if anyone knows of a better chip ;) preferably with a smaller footprint and higher current rating.

I am also looking for a replacement for the `AP2112K-3.3` voltage regulator, since it currently limits the input voltage to max 6V. If we could replace it whit something better, then it would be possible to run the servo on a higher voltage and make the servo faster.

**Reply by Cydget** [reply]
Sorry, I did not look that closely at the schematic, but the datasheet for the TB6612FNG part does have some stipulations to those current ratings. Their graph is not a complete limit, but shows a better estimate at what the rated current actually is. 
![image](https://github.com/adamb314/ServoProject/assets/9152117/76965e34-fcb4-4e20-bdcc-785950fde021)
Its not the clearest, but I would assume that if using both channels, it would be dual-channel operation curve, so at 100% duty cycle aka stalled would be a current rating of 1.8 - 2A. This is also spec'd at 25deg C. Keeping it at that temp may be challenging without a heat sink as the servo will also warm it where ambient temperature is high. Going by the graph, it should be able to operate up to 75deg C. It might be worth throwing a thermal couple on it and measuring when installed in the 3d printed case. 

The other thing for the LDO alt.
One thing I found is that if you do your component search through https://jlcpcb.com/parts/all-electronic-components you can select only their basic parts as a filter. This makes assembly cheaper as they don't charge a setup fee for those basic parts and ordering is cheaper. 

![image](https://github.com/adamb314/ServoProject/assets/9152117/9a98e645-e624-4a9e-8bf8-6dcce3266925)

Looks like AMS1117-5.0 is a possible alt, but has a high voltage drop. If you want to go to a smaller package size there are some other options. 

Personally, I would change the design to have separate nets for the servo and other input with an unpopulated 0 ohm resistor to jump them if needed. That way people can use a larger switching PSU to control many servos, and either regulate that as an input for the micro, or have a secondary supply.
